ci-operator/step-registry/baremetal/lab/bmc/check/baremetal-lab-bmc-check-commands.sh

```diff
@@ -0,0 +1,35 @@
+#!/bin/bash
+
+set -o errexit
+set -o pipefail
+
+if [ -f "${SHARED_DIR}/proxy-conf.sh" ] ; then
+    source "${SHARED_DIR}/proxy-conf.sh"
+fi
+
+proxy="$(<"${CLUSTER_PROFILE_DIR}/proxy")"
+
+if [ "${AGENT_BM_HOSTS_IN_INSTALL_CONFIG}" != "true" ]; then
+  echo "Skipping BMC check step"
+  exit 0
+fi
+
+# wait for bmh to be provisioned
+WAIT="true"
+while [ $WAIT == "true" ]; do
+    sleep 30
+    WAIT=false
+    for bmh in $(oc get bmh -o name -n openshift-machine-api); do
+          if ! [[ $(oc get $bmh -n openshift-machine-api -o jsonpath --template '{.status.provisioning.state}') == "externally provisioned" ]]; then
+                WAIT=true
+          fi
+    done
+done
```

Comment:
If you are waiting for the cluster operator to be ready, you can replace this logic with the following command:
`oc adm wait-for-stable-cluster --minimum-stable-period=2m --timeout=10m`

Reply:
thanks @mhanss , the code is to wait for the baremetal hosts to be provisioned by the operator